### ISSUE TITLE
Comment lines starting with hash (#) - metadata

### ISSUE THREAD

**Issue by sylvainkahane** [UniversalDependencies/docs#856]
Is there a page in the guidelines devoted to the comment lines starting with hash (#) and their normalization?
It seems not. They are mentioned in the pages concerning the [CoNLL-U Format](https://universaldependencies.org/format.html) and its [extension](https://universaldependencies.org/ext-format.html). I think we need some more linguistic guidelines concerning what to put in the comment lines and how to put it. A list of all standardized UD-metadata hashtags would be useful: # text, #sent_id, etc. What is the standard hashtag for text translation, source, genre, speaker id, etc?

**Comment by nschneid** [comment]
Good point. My impression is that beyond `# sent_id` and `# text`, these comment lines are not really standardized. But it would be good to know how they are being used across treebanks (similar to the survey of [MISC attributes](https://universaldependencies.org/misc)).

**Comment by Stormur** [comment]
We are using `# reference` to give the source of the sentence... but this value is itself not standardised (this might be difficult). Anyway, I am also interested in this issue.

**Comment by dan-zeman** [comment]
Agreed that it would be useful. Only `# sent_id`, `# text`, `# newdoc` and `# newpar` are standardized in the UD gudelines, and only the first two are required and checked by the validator (although some other validation rules actually do look at newdoc and newpar if they are present).

There are some others, though, that occur in the examples in the [CoNLL-U format specification](https://universaldependencies.org/format.html), and are documented there as recommendation: `# text_en` (for fluent translation into English; `# text_fr` occurs in one of the examples for translation into French), `# translit` (for transliteration of the sentence text, counterpart of the `Translit` attribute in MISC).

But people obviously use many others, and it is likely that sometimes different teams use different labels for the same thing. This would be nice to harmonize.

**Comment by bguil** [comment]
The page http://tables.grew.fr/?data=meta/META may help for this.

For instance, we can observed that, for English translation, several different metadata are used: `text_en`, `text_eng`, `text[eng]`, `text[eng']`, `Text_eng` and `english_text`.

**Comment by martinpopel** [comment]
Another popular metadata attribute is `global.Entity` - it is used in [UD_English-GUM](https://github.com/UniversalDependencies/UD_English-GUM) and all the [CorefUD](https://ufal.mff.cuni.cz/corefud) treebanks.

BTW, it is not easy to implement an API, which [loads](https://github.com/udapi/udapi-python/blob/b2d1ea858224/udapi/block/read/conllu.py#L37-L81) and [serializes](https://github.com/udapi/udapi-python/blob/b2d1ea858/udapi/block/write/conllu.py#L26-L112) the CoNLL-U comment lines, so that normalized metadata is accessible as properties and their serialization order respects the original file, but has reasonable defaults in case a metadata attribute is added.
